### PR TITLE
Update to new RouterProvider API

### PR DESCRIPTION
**Pull request by brophdawg11**
Update for changes in https://github.com/remix-run/react-router/releases/tag/react-router-dom%406.4.0-pre.15

### CODE REVIEW

**Comment by brophdawg11** [comment]
@austincrim No rush on this - I know you're busy this week!  I'm hoping to live stream your talk tomorrow AM 😄 

This is related to some recent changes we made over in `react-router` to avoid some confusion and nuanced issues (see above linked release notes).  This should get both Vue and Svelte up to date with the new APIs

File: packages/svelte/src/components/RouterProvider.svelte [hunk 1]
```diff
@@ -1,23 +1,13 @@
...
-  export let routes: RouteObject[];
-  export let hydrationData: HydrationState = null;
+  export let router: Router = null;
   export let fallbackElement = null;
 
-  let router = createBrowserRouter({
-    routes,
-    hydrationData,
-  }).initialize();
-
```
Comment (brophdawg11) [inline packages/svelte/src/components/RouterProvider.svelte:20]: This `RouterProvider` now just takes in the instantiated `router`

File: packages/svelte/src/remix-router-svelte.ts [hunk 2]
```diff
@@ -7,12 +7,104 @@ import {
...
 import { getRouteContext, getRouterContext } from "./contexts";
 import { getFormSubmissionInfo, type SubmitOptions } from "./dom";
 
+// Create svelte-specific types from the agnostic types in @remix-run/router to
+// export from remix-router-svelte
+export interface RouteObject extends AgnosticRouteObject {
+  children?: RouteObject[];
+  element?: SvelteComponent | null;
+  errorElement?: SvelteComponent | null;
+}
```
Comment (brophdawg11) [inline packages/svelte/src/remix-router-svelte.ts:29]: In one of the recent releases we got the last react-specific typings out of `@remix-run/router`, so each package now extends `AgnosticRouteObject` with it's own framework-specific typings.

File: packages/svelte/src/remix-router-svelte.ts [hunk 3]
```diff
@@ -7,12 +7,104 @@ import {
...
+
+interface CreateBrowserRouterOpts extends CreateRouterOpts {
+  window?: Window;
+}
+
+interface CreateHashRouterOpts extends CreateRouterOpts {
+  window?: Window;
+}
+
+export function createMemoryRouter(
```
Comment (brophdawg11) [inline packages/svelte/src/remix-router-svelte.ts:63]: 3 utilities for users to create/initialize routers

File: packages/svelte/src/remix-router-svelte.ts (hunk 2 above)
Comment (austincrim) [reply packages/svelte/src/remix-router-svelte.ts:29]: There's a weird nuance with the `svelte` types where you actually need to use `typeof SvelteComponent` for this to work as expected. You'll see the TypeScript errors if you open `reference-app/App.svelte` and switch the `RouteObject` import to use `remix-router-svelte`.

File: packages/svelte/reference-app/App.svelte [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <script lang="ts">
-  import { DataBrowserRouter } from "remix-router-svelte";
+  import { createBrowserRouter, RouterProvider } from "remix-router-svelte";
```
Comment (austincrim) [inline packages/svelte/reference-app/App.svelte:2]: We should switch the `RouteObject` type to import from here as well.

File: packages/svelte/src/remix-router-svelte.ts [hunk 5]
```diff
@@ -7,12 +7,104 @@ import {
...
 import { derived, get, writable, type Readable } from "svelte/store";
 import { getRouteContext, getRouterContext } from "./contexts";
 import { getFormSubmissionInfo, type SubmitOptions } from "./dom";
 
+// Create svelte-specific types from the agnostic types in @remix-run/router to
+// export from remix-router-svelte
+export interface RouteObject extends AgnosticRouteObject {
+  children?: RouteObject[];
+  element?: SvelteComponent | null;
+  errorElement?: SvelteComponent | null;
```
[the remark this answers is not in the document]
Comment (brophdawg11) [reply packages/svelte/src/remix-router-svelte.ts:28]: Ah - right yeah I'll remove for now with a comment 👍

**Comment by brophdawg11** [comment]
Thanks for the feedback!  Assuming all the CI passes I'll get this merged and then we can each release the packages whenever

**Pull request merged**